test/__init__.py

```diff
@@ -0,0 +1,79 @@
+
+import unittest
+import binascii
+import difflib
+import struct
+import array
+
+import Xlib.protocol.event
+import Xlib.protocol.rq
+
+
+class CmpArray(object):
+
+    def __init__(self, *args, **kws):
+        self.array = array.array(*args, **kws)
+
+    def __len__(self):
+        return len(self.array)
+
+    def __getitem__(self, key):
+        if isinstance(key, slice):
+            x = key.start
+            y = key.stop
+            return list(self.array[x:y])
+        else:
+            return self.array[key]
+
+    def __getattr__(self, attr):
+        return getattr(self.array, attr)
+
+    def __eq__(self, other):
+        return self.array.tolist() == other
+
+Xlib.protocol.rq.array = CmpArray
+
+
+class DummyDisplay(object):
+
+    def get_resource_class(self, x):
+        return None
+
+    event_classes = Xlib.protocol.event.event_class
+
+
+class TestCase(unittest.TestCase):
+
+    def assertBinaryEqual(self, bin1, bin2):
+        if bin1 == bin2:
+            return
+        self.fail('binary contents differ:\n' + bindiff(bin1, bin2))
```

Comment:
> Why not just if bin1 != bin2: self.fail(...)? It avoids explicit empty return and code is more compact.

OK, amended.

> I would also prefer having PEP8 compatible names where possible (assert_binary_equal instead of assertBinaryEqual for method names).

This was in keeping with the style used by unittest for assert methods, something PEP8 mention is preferred (consistency over strict adherence).